streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -282,7 +284,22 @@ public <K, V> void send(final String topic,
                     topicProducedSensor.record(bytesProduced, context.currentSystemTimeMs());
                 }
             } else {
-                recordSendError(topic, exception, serializedRecord, context, processorNodeId);
+                final RecordContext recordContext = context.recordContext();
```

Comment:
Isn't the issue here that `context` might be `null`?
See https://github.com/apache/kafka/blob/a4ea9aec73a3da470d079bad82d92810cac49d55/streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java#L143